src/quickpicks/localResources.ts

```diff
@@ -79,62 +74,62 @@ export async function localResourcesQuickPick(
-      quickpick.selectedItems = [kafkaItem, ...items];
-      window.showInformationMessage("Kafka must be available before starting Schema Registry.");
-    }
-  });
-
-  // user clicked one of the secondary buttons to the right of a quickpick item
-  quickpick.onDidTriggerItemButton(
-    async (event: { button: QuickInputButton; item: QuickPickItem }) => {
-      quickpick.hide();
-      if (event.button.tooltip?.includes(LocalResourceKind.Kafka)) {
+  // Create the quickpick using our enhanced utility
+  const quickpick = createEnhancedQuickPick(items, {
+    title,
+    placeHolder: placeholder,
+    ignoreFocusOut: true,
+    canSelectMany: true,
+    selectedItems: starting && !kafkaAvailable ? [kafkaItem] : [],
+    onSelectionChange: (items, quickPick) => {
+      if (
+        !starting &&
+        quickPick.items.includes(schemaRegistryItem) &&
+        items.includes(kafkaItem) &&
+        !items.includes(schemaRegistryItem)
+      ) {
+        // if the user is attempting to stop Kafka and not Schema Registry, ensure Schema Registry is
+        // selected
+        quickPick.selectedItems = [...items, schemaRegistryItem];
+        window.showInformationMessage("Schema Registry must be stopped when stopping Kafka.");
+      } else if (
+        starting &&
+        quickPick.items.includes(kafkaItem) &&
+        items.includes(schemaRegistryItem) &&
+        !items.includes(kafkaItem)
+      ) {
+        // if the user is attempting to start Schema Registry and not Kafka, ensure Kafka is selected
+        quickPick.selectedItems = [kafkaItem, ...items];
+        window.showInformationMessage("Kafka must be available before starting Schema Registry.");
+      }
+    },
+    onItemButtonClicked: async ({ button, quickPick }) => {
+      quickPick.hide();
+      if (button.tooltip?.includes(LocalResourceKind.Kafka)) {
         // open Settings and filter to the Kafka image repo+tag settings
         commands.executeCommand(
           "workbench.action.openSettings",
           `@id:${LOCAL_KAFKA_IMAGE} @id:${LOCAL_KAFKA_IMAGE_TAG}`,
         );
-      } else if (event.button.tooltip?.includes(LocalResourceKind.SchemaRegistry)) {
+      } else if (button.tooltip?.includes(LocalResourceKind.SchemaRegistry)) {
         // open Settings and filter to the Schema Registry image repo+tag settings
         commands.executeCommand(
           "workbench.action.openSettings",
           `@id:${LOCAL_SCHEMA_REGISTRY_IMAGE} @id:${LOCAL_SCHEMA_REGISTRY_IMAGE_TAG}`,
         );
       }
     },
-  );
+  });
+
+  // Show the QuickPick
+  quickpick.show();
 
   // user clicked "Ok" or hit Enter to accept the selected items, if any
   const selectedItems: QuickPickItem[] = [];
   quickpick.onDidAccept(() => {
     logger.debug("selected items", { items: JSON.stringify(quickpick.selectedItems) });
     selectedItems.push(...quickpick.selectedItems);
     quickpick.hide();
-    return;
   });
 
   // block until the quickpick is hidden
```

Comment:
And we should probably remove this block as well if the new function blocks by default